src/fides/api/models/privacy_preference.py

```diff
@@ -388,3 +514,130 @@ def get_preference_for_notice_and_fides_user_device(
             )
             .first()
         )
+
+
+class LastServedNotice(LastSavedMixin, Base):
+    """Stores the last time a notice was served for a given user
+
+    Also consolidates serving notices among various user identities.
+    """
+
+    served_notice_history_id = Column(
+        String, ForeignKey(ServedNoticeHistory.id), nullable=False, index=True
+    )
+
+    __table_args__ = (
+        UniqueConstraint(
+            "provided_identity_id",
+            "privacy_notice_id",
+            name="last_served_identity_privacy_notice",
+        ),
+        UniqueConstraint(
+            "fides_user_device_provided_identity_id",
+            "privacy_notice_id",
+            name="last_served_fides_user_device_identity_privacy_notice",
+        ),
+    )
+
+    # Relationships
+    served_notice_history = relationship(
+        ServedNoticeHistory, cascade="delete, delete-orphan", single_parent=True
+    )
+
+    @property
+    def served_latest_version(self) -> bool:
+        """Returns True if the user was last served the latest version of this Notice"""
+        return (
+            self.privacy_notice.privacy_notice_history_id
+            == self.privacy_notice_history_id
+        )
+
+    @classmethod
+    def get_last_served_for_notice_and_fides_user_device(
+        cls,
+        db: Session,
+        fides_user_provided_identity: ProvidedIdentity,
+        privacy_notice: PrivacyNotice,
+    ) -> Optional[LastServedNotice]:
+        """Retrieves the LastServedNotice record for the user with the given identity
+        for the given notice"""
+        return (
+            db.query(LastServedNotice)
+            .filter(
+                LastServedNotice.fides_user_device_provided_identity_id
+                == fides_user_provided_identity.id,
+                LastServedNotice.privacy_notice_id == privacy_notice.id,
+            )
+            .first()
+        )
+
+
+def upsert_last_saved_record(
+    db: Session,
+    created_historical_record: Union[PrivacyPreferenceHistory, ServedNoticeHistory],
+    current_record_class: Union[Type[CurrentPrivacyPreference], Type[LastServedNotice]],
+    privacy_notice_history: PrivacyNoticeHistory,
+    current_record_data: Dict[str, str],
+) -> Union[CurrentPrivacyPreference, LastServedNotice]:
```

Comment:
This more or less existed previously, just abstracting to share with upserting the new LastServedNotice records.   Both of these record types just store the last time consent was saved or served for the given notice across all known identities, to make this more easily queryable for the frontend when building banners, overlays, and privacy centers for display.